### PR TITLE
upgrade theano 1.0

### PR DESCRIPTION
**Pull request by cjcchen**
theano has been upgraded to 1.0

### CODE REVIEW

**Comment by ryanjulian** [comment]
@cjcchen can you confirm that this doesn't break anything?

**Comment by ryanjulian** [comment]
fixes #17

**Comment by cjcchen** [comment]
I have checked it from the website and run the all the examples in the example folder. It seems only the Lasagne library has to be updated.

**Comment by cjcchen** [comment]
In the official document,  one change is related to our library "Changed grad() method to L_op() in ops that need the outputs to compute gradient". But it seems the grad function in rllab works fine after I  upgraded the theano to verioin 1.0.

**Review by eric-heiden — DISMISSED** [review]

File: environment.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies:
     - pygame
     - matplotlib
     - pandas
+    - mkl-service
```
Comment (ryanjulian) [inline environment.yml]: please pin to a specific version

File: environment.yml [hunk 2]
```diff
@@ -25,8 +26,8 @@ dependencies:
         - pyzmq
         - cached_property
         - cloudpickle
-        - git+https://github.com/Theano/Theano.git@adfe319ce6b781083d8dc3200fb4481b00853791#egg=Theano
-        - git+https://github.com/neocxi/Lasagne.git@484866cf8b38d878e92d521be445968531646bb8#egg=Lasagne
+        - git+https://github.com/Lasagne/Lasagne.git#egg=Lasagne
+        - git+https://github.com/Theano/Theano.git#egg=Theano
```
Comment (ryanjulian) [inline environment.yml]: please pin to a specific version or git hash/release tag.

http://deeplearning.net/software/theano_versions/dev/install_ubuntu.html states "The latest stable version of Theano is 1.0.1 (tagged with rel-1.0.1)."

File: environment.yml [hunk 3]
```diff
@@ -25,8 +26,8 @@ dependencies:
         - pyzmq
         - cached_property
         - cloudpickle
-        - git+https://github.com/Theano/Theano.git@adfe319ce6b781083d8dc3200fb4481b00853791#egg=Theano
-        - git+https://github.com/neocxi/Lasagne.git@484866cf8b38d878e92d521be445968531646bb8#egg=Lasagne
+        - git+https://github.com/Lasagne/Lasagne.git#egg=Lasagne
```
Comment (ryanjulian) [inline environment.yml]: please pin to a specific version or git hash/release tag

**Review by ryanjulian — CHANGES_REQUESTED** [review]
We need to pin versions for large dependencies (really, all dependencies, but it is most important for large ones).

**Pull request closed without merging**